transports/quic/src/endpoint.rs

```diff
@@ -0,0 +1,471 @@
+use crate::crypto::{Crypto, CryptoConfig};
+use crate::muxer::QuicMuxer;
+use crate::{QuicConfig, QuicError};
+use futures::channel::{mpsc, oneshot};
+use futures::prelude::*;
+use quinn_proto::crypto::Session;
+use quinn_proto::generic::{ClientConfig, ServerConfig};
+use quinn_proto::{
+    ConnectionEvent, ConnectionHandle, DatagramEvent, EcnCodepoint, EndpointEvent, Transmit,
+};
+use std::collections::{HashMap, VecDeque};
+use std::io::IoSliceMut;
+use std::mem::MaybeUninit;
+use std::net::SocketAddr;
+use std::pin::Pin;
+use std::sync::Arc;
+use std::task::{Context, Poll};
+use std::time::Instant;
+use udp_socket::{RecvMeta, SocketType, UdpCapabilities, UdpSocket, BATCH_SIZE};
+
+/// Message sent to the endpoint background task.
+#[derive(Debug)]
+enum ToEndpoint<C: Crypto> {
+    /// Instructs the endpoint to start connecting to the given address.
+    Dial {
+        /// UDP address to connect to.
+        addr: SocketAddr,
+        /// The remotes public key.
+        public_key: C::PublicKey,
```

Comment:
Do I understand correctly that one can thus not dial a remote without their `PeerId`?